docs/guides/runners.md

```diff
@@ -0,0 +1,114 @@
+# Running benchmarks with the benchmark runner
+
+The `BecnhmarkRunner` is `nnbench`'s abstract interface to collect and execute benchmarks.
+You can either use it directly or instantiate a derivative class to override selected methods and implement you own logic.
+
+
+##  The abstract `BenchmarkRunner`  class
+Let's first instantiate and then walk through the abstract class.
+
+```python
+from nnbench import BenchmarkRunner
+
+runner = BenchmarkRunner()
+```
+
+Now we can use the `BenchmarkRunner.collect` method to collect benchmarks from files or directories.
+Lets assume we have the following benchmark setup:
+```python
+# ./performance/model_benchmark.py
+import nnbench
+import numpy as np
+from sklearn import base, metrics
+
+@nnbench.benchmark
+def accuracy(model: base.BaseEstimator, X_test: np.ndarray, y_test: np.ndarray) -> float:
+    y_pred = model.predict(X_test)
+    accuracy = metrics.accuracy_score(y_test, y_pred)
+    return accuracy
+```
+
+```python
+# ./model_info/inference_resources.py
+import nnbench
+import numpy as np
+import time
+from sklearn import base
+from memory_profiler import memory_usage
+
+@nnbench.benchmark(tags=("inference-resources",))
+def avg_inference_time(model: base.BaseEstimator, X_test: np.ndarray, steps: int) -> float:
+    total_time = 0.0
+
+    for _ in range(steps):
+        start_time = time.time()
+        model.predict(X_test)
+        end_time = time.time()
+        total_time += end_time - start_time
+
+    average_time = total_time / steps
+    return average_time
+
+@nnbench.benchmark(tags=("inference-resources",))
+def memory_usage_during_inference(model: base.BaseEstimator, X_test: np.ndarray) -> float:
+    def predict():
+        model.predict(X_test)
+
+    mem_usage = memory_usage(predict, max_usage=True)
+    return mem_usage
+```
+
+```python
+# ./model_info/storage_resources.py
+import nnbench
+import numpy as np
+from sklearn import base
+
+@nnbench.benchmark
+def model_size_mb(model: base.BaseEstimator) -> float:
+    total_size_bytes = 0
+
+    for attr, value in model.__dict__.items():
+        if isinstance(value, np.ndarray):
+            total_size_bytes += value.nbytes
+        elif isinstance(value, list) or isinstance(value, dict):
+            for item in value:
```

Comment:
For a `dict` value, this does not do what you think it does: `dict.__iter__` gives an iterator over the keys, not the values (which you are looking for here).